platform/machine/aws/machine.go

```diff
@@ -113,6 +124,20 @@ func (am *machine) saveConsole(origConsole string) error {
 		return true, nil
 	})
 	if err != nil {
+		if origConsole != "" {
+			if writeErr := am.writeConsole(origConsole); writeErr != nil {
```

Comment:
This would be cleaner as:

```go
err := fmt.Errorf("retrieving console output of %v: %v", am.ID(), err)
if origConsole != "" {
    plog.Warning(err)
    // fall through, to save what we can
} else {
    return err
}
```

Then the writing code can remain unchanged, and you don't need `writeConsole`.  I don't know that it's necessary to distinguish partial output due to timeout from partial output due to other kinds of failures.

Reply:
Agreed, updated.